packages/client/src/graphql/Foretold__GraphQL__GetAgentMeasurements.re

```diff
@@ -0,0 +1,183 @@
+type measurable = {
+  id: string,
+  name: string,
+  state: Context.Primary.MeasurableState.t,
+  stateUpdatedAt: option(MomentRe.Moment.t),
+  expectedResolutionDate: option(MomentRe.Moment.t),
+};
+
+// @todo:
+type connection('a) =
+  option({
+    .
+    "edges": option(Js.Array.t(option({. "node": option('a)}))),
+    "pageInfo": {
+      .
+      "endCursor": option(string),
+      "hasNextPage": bool,
+      "hasPreviousPage": bool,
+      "startCursor": option(string),
+    },
+    "total": option(int),
+  });
+
+type node = {
+  id: string,
+  relevantAt: option(MomentRe.Moment.t),
+  competitorType: Context.Primary.CompetitorType.t,
+  description: option(string),
+  createdAt: MomentRe.Moment.t,
+  taggedMeasurementId: option(string),
+  measurable: option(measurable),
+  value: MeasurementValue.graphQlResult,
+};
+
+type agent = {measurements: connection(node)};
+
+let unpackEdges = (a: connection('a)): array('a) => {
+  a
+  |> Rationale.Option.fmap(b =>
+       b##edges |> E.A.O.defaultEmpty |> E.A.O.concatSome
+     )
+  |> E.A.O.defaultEmpty
+  |> Array.map(e => e##node)
+  |> E.A.O.concatSome;
+};
+
+module Query = [%graphql
+  {|
+    query getAgentMeasurements ($id: String!, $first: Int, $last: Int, $after: String, $before: String) {
+        agent:
+        agent(id: $id) @bsRecord{
+            measurements: Measurements (first: $first, last: $last, after: $after, before: $before) {
```

Comment:
Instead of doing this, can you use the ``measurements`` query, where you pass in the ``agentId`` (instead of using the agent > measurements query)? That would allow us to filter on more things in the future, unless we want to add the same filtering to this subquery.

Reply:
Good idea, will do.

Reply:
Fixed.